commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/CommonsLangPortedFractionTest.java

```diff
@@ -0,0 +1,1144 @@
+        assertThrows(ArithmeticException.class, () -> smallest.divide(smallest.reciprocal())); // Should overflow
+
+        final Fraction negative = Fraction.of(1, -Integer.MAX_VALUE);
+        assertThrows(ArithmeticException.class, () -> negative.divide(negative.reciprocal())); // Should overflow
+    }
+
+    @Test
+    public void testEquals() {
+        Fraction f1 = null;
+        Fraction f2 = null;
+
+        f1 = Fraction.of(3, 5);
+        assertNotEquals(null, f1);
+        assertNotEquals(f1, new Object());
+        assertNotEquals(f1, Integer.valueOf(6));
+
+        f1 = Fraction.of(3, 5);
+        f2 = Fraction.of(2, 5);
+        assertNotEquals(f1, f2);
+        assertEquals(f1, f1);
+        assertEquals(f2, f2);
+
+        f2 = Fraction.of(3, 5);
+        assertEquals(f1, f2);
+
+        f2 = Fraction.of(6, 10);
+        assertEquals(f1, f2);
+    }
+
+    @Test
+    public void testHashCode() {
+        final Fraction f1 = Fraction.of(3, 5);
+        Fraction f2 = Fraction.of(3, 5);
+
+        assertEquals(f1.hashCode(), f2.hashCode());
+
+        f2 = Fraction.of(2, 5);
+        assertTrue(f1.hashCode() != f2.hashCode());
+
+        f2 = Fraction.of(6, 10);
+        assertEquals(f1.hashCode(), f2.hashCode());
+    }
+
+    @Test
+    public void testCompareTo() {
+        Fraction f1 = null;
+        Fraction f2 = null;
+
+        f1 = Fraction.of(3, 5);
+        assertEquals(0, f1.compareTo(f1));
+
+        final Fraction fr = f1;
+        assertThrows(NullPointerException.class, () -> fr.compareTo(null));
+
+        f2 = Fraction.of(2, 5);
+        assertTrue(f1.compareTo(f2) > 0);
+        assertEquals(0, f2.compareTo(f2));
+
+        f2 = Fraction.of(4, 5);
+        assertTrue(f1.compareTo(f2) < 0);
+        assertEquals(0, f2.compareTo(f2));
+
+        f2 = Fraction.of(3, 5);
+        assertEquals(0, f1.compareTo(f2));
+        assertEquals(0, f2.compareTo(f2));
+
+        f2 = Fraction.of(6, 10);
+        assertEquals(0, f1.compareTo(f2));
+        assertEquals(0, f2.compareTo(f2));
+
+        /*
+         *  Removed as not supported in numbers.
+         *
+         *  f2 = Fraction.of(-1, 1, Integer.MAX_VALUE);
+         *  assertTrue(f1.compareTo(f2) > 0);
+         *  assertEquals(0, f2.compareTo(f2));
+         */
+    }
+
+    @Test
+    public void testToString() {
+        Fraction f = null;
+
+        f = Fraction.of(3, 5);
+        final String str = f.toString();
+        assertEquals("3 / 5", str);
+        assertEquals(str, f.toString());
+
+        f = Fraction.of(7, 5);
+        assertEquals("7 / 5", f.toString());
+
+        f = Fraction.of(4, 2);
+        assertEquals("2", f.toString());
+
+        f = Fraction.of(0, 2);
+        assertEquals("0", f.toString());
+
+        f = Fraction.of(2, 2);
+        assertEquals("1", f.toString());
+
+        /*
+         *  Removed as not supported in numbers.
+         *
+         *  f = Fraction.of(Integer.MIN_VALUE, 0, 1);
```

Comment:
Possibly replace with:
```java
f = Fraction.of(Integer.MIN_VALUE);
```

Reply:
@aherbert done.

Reply:
@aherbert  actually, what they want here is `f = Fraction.of(-1).add(Fraction.of(-1, Integer.MAX_VALUE));`

Reply:
I thought this was creating `X Y/Z` with `Y=0` so effectively just creating a whole number.

Reply:
OK. This refers to the next line so you are correct.

Reply:
@aherbert this sugests that `X Y/Z` parser thing is really confusing.
And good to not have that parser in commons-numbers.